src/comm.cc

```diff
@@ -1275,30 +1275,38 @@ commHandleWriteHelper(void * data)
     assert(clientInfo);
     assert(clientInfo->hasQueue());
     assert(clientInfo->hasQueue(queue));
-    assert(!clientInfo->selectWaiting);
     assert(clientInfo->eventWaiting);
     clientInfo->eventWaiting = false;
 
     do {
-        // check that the head descriptor is still relevant
-        const int head = clientInfo->quotaPeekFd();
-        Comm::IoCallback *ccb = COMMIO_FD_WRITECB(head);
+        if (clientInfo->writeOrQuotaDequeue())
+            return;
```

Comment:
The proposed writeOrQuotaDequeue() interface is 

* misleading -- the true result does not mean "wrote or dequeued" as one would expect
* unnecessary complex -- the caller can check `selectWaiting` if it cares about that state

I suggest solving this problem by simplifying the new method and moving its return result logic into the caller:

```suggestion
        clientInfo->writeOrDequeue();
        if (clientInfo->selectWaiting)
            return;
```

Reply:
Done.